### ISSUE TITLE
Cursors are broken in 3.6.0

### ISSUE THREAD

**Issue by baskerville** [facebook/immutable-js#283]
The following example:
```
var Immutable = require("immutable"),
    Cursor = require("immutable/contrib/cursor");

var jsData = [{a: 1}, {a: 2}, {a: 3}],
    data = Immutable.fromJS(jsData),
    cursor = Cursor.from(data);

cursor.forEach(function (c) {
	console.log(c);
});
```

works fine with *3.5.0*, but produce the following error with *3.6.0*:
```
RangeError: Maximum call stack size exceeded
```

**Comment by leebyron** [comment]
Thanks for the red-alert! I'll get this fixed pronto

**Comment by leebyron** [comment]
Fixed in 3.6.1
